packages/yasqe/src/sparql.ts

```diff
@@ -50,46 +49,56 @@ export function getAjaxConfig(
 }
 
 export async function executeQuery(yasqe: Yasqe, config?: YasqeAjaxConfig): Promise<any> {
-  var req: superagent.SuperAgentRequest;
+  const queryStart = Date.now();
   try {
     yasqe.emit("queryBefore", yasqe, config);
-    getAjaxConfig(yasqe, config);
     const populatedConfig = getAjaxConfig(yasqe, config);
     if (!populatedConfig) {
-      //nothing to query
-      return;
+      return; // Nothing to query
+    }
+    const abortController = new AbortController();
+    const fetchOptions: RequestInit = {
+      method: populatedConfig.reqMethod,
+      headers: {
+        Accept: populatedConfig.accept,
+        ...(populatedConfig.headers || {}),
+        "Content-Type": populatedConfig.reqMethod === "POST" ? "application/x-www-form-urlencoded" : "",
+      },
+      credentials: populatedConfig.withCredentials ? "include" : "same-origin",
+      signal: abortController.signal,
+    };
+    const searchParams = new URLSearchParams();
+    for (const key in populatedConfig.args) {
+      const value = populatedConfig.args[key];
+      if (Array.isArray(value)) {
+        value.forEach((v) => searchParams.append(key, v));
+      } else {
+        searchParams.append(key, value);
+      }
     }
-    var queryStart = Date.now();
-
     if (populatedConfig.reqMethod === "POST") {
-      req = superagent.post(populatedConfig.url).type("form").send(populatedConfig.args);
+      fetchOptions.body = searchParams.toString();
     } else {
-      req = superagent.get(populatedConfig.url).query(populatedConfig.args);
+      populatedConfig.url += `?${searchParams.toString()}`;
     }
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    req.accept(populatedConfig.accept).set(populatedConfig.headers || {});
-
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    if (populatedConfig.withCredentials) req.withCredentials();
-    yasqe.emit("query", req, populatedConfig);
-    return await req.then(
-      (result) => {
-        yasqe.emit("queryResponse", result, Date.now() - queryStart);
-        yasqe.emit("queryResults", result.body, Date.now() - queryStart);
-        return result.body;
-      },
-      (e) => {
-        if (e instanceof Error && e.message === "Aborted") {
-          //The query was aborted. We should not do or draw anything
-        } else {
-          yasqe.emit("queryResponse", e, Date.now() - queryStart);
-        }
-        yasqe.emit("error", e);
-        throw e;
-      }
-    );
+    const request = new Request(populatedConfig.url, fetchOptions);
+    yasqe.emit("query", request, abortController);
+    const response = await fetch(request);
+    const result = await response.json();
+    if (!response.ok) {
+      throw new Error(result || response.statusText);
+    }
+    yasqe.emit("queryResponse", result, Date.now() - queryStart);
```

Comment:
Should it be `response` instead of `result` here? 🤔 
I mean, currently it's returning `result` and then `result.body` for `queryResults`.

Reply:
Good question. In the `YASQE` interface the `queryResponse` event was defined as

```ts
on("queryResponse", (instance: Yasqe, req: superagent.SuperAgentRequest, duration: number))

// And triggered:
private handleQueryResponse(_yasqe: Yasqe, _response: superagent.SuperAgentRequest, duration: number) {
    this.lastQueryDuration = duration;
    this.req = undefined;
    this.updateQueryButton();
}
```

So I instinctively replaced it by a `Request`, but I think in superagent there is no Response type, everything goes into the SuperAgentRequest. With fetch it would make much more sense to send the response back, and we could also emit the `queryResponse` before `await response.json()`. 

I made the changes so now the `queryResponse` event takes a `Response` instead of a `Request`

For `queryResult` we are good because `const result = await response.json()` already gets the body as an `any` dict (which is what was the previous behavior)